lib/util/lifted/influx/coordinator/statement_executor.go

```diff
@@ -1402,6 +1407,130 @@ func (e *StatementExecutor) executeShowMeasurementsStatement(q *influxql.ShowMea
 	}, seq)
 }
 
+func (e *StatementExecutor) executeShowMeasurementsDetailStatement(stmt *influxql.ShowMeasurementsDetailStatement, ctx *query.ExecutionContext, seq int) error {
+	if stmt.Database == "" {
+		return coordinator.ErrDatabaseNameRequired
+	}
+	var mms influxql.Measurements
+	if stmt.Source != nil {
+		mms = influxql.Measurements{stmt.Source.(*influxql.Measurement)}
+	}
+
+	measurements, err := e.MetaClient.MatchMeasurements(stmt.Database, mms)
+	if err != nil {
+		return err
+	}
+
+	var blank2Nil = func(s string) string {
+		// not return a empty string, return "<nil>" instead
+		if s == "" {
+			return "<nil>"
+		}
+		return s
+	}
+
+	// sliceFirstColumn gets all shard keys and indexes from the returned models.Row.Values array, then returns a sorted slice.
+	// sk: true if the values are shard keys.
+	var sliceFirstColumn = func(values [][]interface{}, sk bool) (ret []string) {
+		ret = make([]string, 0, len(values))
+		if sk {
+			for i := range values {
+				ret = append(ret, values[i][0].([]string)...)
```

Comment:
give bounds checking in place, how to do if `values[i][0]` is nil? if the interface assertion does not meet the expectation?

Reply:
Fixed. Thanks for the suggestion.